src/main/java/org/spongepowered/common/mixin/api/mcp/entity/monster/SpellcastingIllagerEntityMixin_API.java

```diff
@@ -51,4 +54,17 @@ public void setCastingSpell(boolean castSpell) {
 
         // i509 -> TODO: Figure out how to get the casting time from UseSpellGoal. Yes Mojang made spells a goal and their oh so exposed enum SpellType does not refer to the goals.
     }
+
+    @Override
+    protected Set<Value.Immutable<?>> api$getVanillaValues() {
+        final Set<Value.Immutable<?>> vanillaValues = super.api$getVanillaValues();
+
+        vanillaValues.add(this.castingTime().asImmutable());
+
+        if (this.activeSpell == SpellcastingIllagerEntity.SpellType.NONE) {
```

Comment:
Providers will know to filter on `SpellType.NONE`, so just call the api method for the current spell and the `ifPresent` will know if the value is empty or not.